agent/src/jvmMain/kotlin/com/epam/drill/agent/websocket/WsClientConnector.kt

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2020 - 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.agent.websocket
+
+import java.net.URI
+import javax.websocket.ClientEndpointConfig
+import javax.websocket.ContainerProvider
+import javax.websocket.WebSocketContainer
+import org.eclipse.jetty.websocket.jsr356.ClientContainer
+import mu.KotlinLogging
+import com.epam.drill.agent.configuration.WsConfiguration
+import com.epam.drill.common.agent.configuration.HEADER_AGENT_CONFIG
+
+private const val HEADER_CONTENT_ENCODING = "Content-Encoding"
+
+class WsClientConnector(
+    private val uri: URI,
+    private val endpoint: WsClientEndpoint
+) : ClientEndpointConfig.Configurator() {
+
+    private val logger = KotlinLogging.logger {}
+    private val container = ContainerProvider.getWebSocketContainer().apply(::configureWebSocketContainer)

```

Comment:
Should we just move `ContainerProvider.getWebSocketContainer()` to `configureWebSocketContainer` function?